### PR TITLE
Ch5 gxd

### PR DESCRIPTION
**Pull request by GrahamDavidson**

### CODE REVIEW

File: src/cypher.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+
+public class cypher {
+  public static void main (String args[]){
+  	
+  }
```
Comment (Spinodal) [inline src/cypher.java:6]: so effective it encrypted the source code!

File: src/average.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+
+public class average {
```
Comment (SandyThomson) [inline src/average.java:3]: In Java, the generally accepted way to name your classes (both in terms of filename, and the classname in the file), is to start with a capital letter (so this class and filename would be `Average(.java)` ).

Methods and variables generally start with lower case characters. FYI.

File: src/minMax.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+
+
+public class minMax {
+  public static void main(String args[]){
+  	int nums[] = {99, -10, 100123, 18, -978, 5623, 463, -9, 287, 49};
+  	int min    = nums[0];
+  	int	max    = nums[0];
+  	
+  	for (int val : nums){
```
Comment (SandyThomson) [inline src/minMax.java:9]: All good!

File: src/sortStrings.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    // Display original array
+    System.out.print( "Original array is:");
+    for ( String string : data) System.out.print ( " " + string );
+    System.out.println();
+    
+    // This is the Bubble Sort
+    for ( a = 1; a < size ; a++ ) {
+      for ( b = size - 1 ; b >= a ; b -- )  {
+      	if(data[b-1].compareTo( data[b] ) > 0 ){
```
Comment (SandyThomson) [inline src/sortStrings.java:22]: All good. It looks a bit odd (IMO) when you start mixing in capital letters, but this works.

File: src/minMax.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  		if (val < min) min = val;
+  		if (val > max) max = val;
```
Comment (mocathe1st) [inline src/minMax.java:11]: This is a candidate for the ternary operator, so:
    val = val < min ? min : val;
Although the way you've done it is probably better in this case tbh

File: src/sortStrings.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
+
+
+/*
+ * Sort an array of strings
+ */
+
+public class sortStrings {
```
Comment (mocathe1st) [inline src/sortStrings.java:7]: Like Sandy's said elsewhere, general Java convention is to start class names with a capital. The camel casing is good though and is generally used in Java for names (always catches me out in Perl, as I tend to camel case everything :cry: )

**Pull request closed without merging**